src/service/standard-actions-impl.js

```diff
@@ -205,6 +216,92 @@ export class StandardActions {
     throw user().createError('Unknown AMP action ', method);
   }
 
+  /**
+   * Handles the copy to clipboard action
+   * @param {!./action-impl.ActionInvocation} invocation
+   * @return {!null}
+   */
+  handleCopy_(invocation) {
+    /** @enum {string} */
+    const CopyEvents = {
+      COPY_ERROR: 'copy-error',
+      COPY_SUCCESS: 'copy-success',
+    };
+    let eventResult;
+    const {args, node} = invocation;
+    const win = getWin(node);
+
+    let textToCopy;
+    if (invocation.tagOrTarget === 'AMP') {
+      /**
+       * Copy Static Text
+       *  Example: AMP.copy(text='TextToCopy');
+       */
+      textToCopy = args['text'].trim();
+    } else {
+      /**
+       * Copy Target Element Text
+       *  Example: targetId.copy();
+       */
+      const target = devAssertElement(invocation.node);
+      textToCopy = (target.value ?? target.textContent).trim();
+    }
+
+    /**
+     * Raises a status event for copy task result
+     * @param {*} eventName
+     * @param {*} eventResult
+     * @param {*} invocation
+     */
+    const triggerEvent = function (eventName, eventResult, invocation) {
+      const eventValue = /** @type {!JsonObject} */ ({
+        data: /** @type {!JsonObject} */ {type: eventResult},
+      });
+      const copyEvent = createCustomEvent(win, `${eventName}`, eventValue);
+
+      const action_ = Services.actionServiceForDoc(invocation.caller);
+      action_.trigger(
+        invocation.caller,
+        eventName,
+        copyEvent,
+        ActionTrust_Enum.HIGH
+      );
+    };
+
+    /**
+     * Trigger Event based on copy action
+     *  - If content got copied to the clipboard successfully, it will
+     *    fire `copy-success` event with data type `success`.
+     *  - If there's any error in copying, it will
+     *    fire `copy-error` event with data type `error`.
+     *  - If browser is not supporting the copy function/action, it
+     *    will fire `copy-error` event with data type `browser`.
+     *
+     *  Example: <button on="tap:AMP.copy(text='Hello AMP');copy-success:copied.show()">Copy</button>
+     */
+    let eventName = CopyEvents.COPY_ERROR;
+    if (isCopyingToClipboardSupported(win.document)) {
+      copyTextToClipboard(
+        win,
+        textToCopy,
+        () => {
+          eventName = CopyEvents.COPY_SUCCESS;
+          eventResult = 'success';
+          triggerEvent(eventName, eventResult, invocation);
+        },
+        () => {
+          eventResult = 'error';
+          triggerEvent(eventName, eventResult, invocation);
+        }
+      );
+    } else {
+      eventResult = 'browser';
```

Comment:
Can you explain the browser event result? Should this be `unsupported` instead?

Reply:
Yes, sorry the confusing name. I have renamed it to `unsupported` instead.